README.md

```diff
@@ -1,27 +1,67 @@
 # Random Coin Toss
 
-## Overview
+While both are backed by Flow's Random Beacon,
+it is important for developers to mindfully choose between `revertibleRandom`
+or seeding their own PRNG utilizing the `RandomBeaconHistory` smart contract:
 
-The contracts contained in this repo demonstrate how to use Flow's onchain randomness safely - safe randomness here
-meaning non-revertible randomness.
+- Under the hood, the FVM also just instantiates a PRNG for each transaction that `revertibleRandom` draws from. 
```

Comment:
```suggestion
- Under the hood, the FVM instantiates a PRNG for each transaction that `revertibleRandom` draws from. 
```

Reply:
What is the source of randomness for a call to revertibleRandom at runtime? Is it some previous block's source of randomness? I ask because it seems there are two risks with using the method alone, please correct my understanding if it's off:

1. The source of randomness when calling revertibleRandom will be source known to the deterministic environment and/or already available to the execution environment (e.g. a previous source of randomness archived in RandomBeaconHistory)
2. The ability to post-condition transaction execution when called by untrusted parties

If 1/ is true, I think it would be worth adding a note about it in this doc

Reply:
> What is the source of randomness for a call to revertibleRandom at runtime?

It is a source of randomness of the block where the transaction was included, and not the source of randomness of any past block. The transaction runtime has no access to that source, it is known only by the FVM and used to seed an internal PRNG. The PRNG runs within FVM and only the generated randoms are passed to the transaction runtime. 
Moreover, that PRNG is a special one (called Crypto-Secure, unlike the xorshift for instance), so that given as many PRNG outputs as we want, it's not possible to retrieve the source of randomness, or compute future random numbers.  This way the transaction runtime can never guess the future randoms (only FVM can).

Reply:
As Tarak explained, point 1 is not true:
* The source of the randomness for a transaction `T` is completely unknown until the block containing `T` has been constructed and certified by consensus. Only at that point, the result of the transaction becomes "knowable", i.e. all the required inputs of the transaction are available. 
* The randomness is a system level function that any code running in the "user space" has no access to. It is similar to a hardware TPM in your computer: software can interact with it via the exposed APIs but it cannot access the TPMs internal state. Same here: smart contracts can access Flow's source of randomness by calling `revertibleRandom` but they cannot access the random number generator backing `revertibleRandom`.

Reply:
@sisyphusSmiling what do you think, should we go into more detail? Or is it fine the way the doc is because 1. is no concern (?)

Reply:
Thanks for the clarification! The details are certainly interesting and impressive, but I think they're best left for the in-depth docs/engineering-focussed blog post that's been mentioned elsewhere.